### ISSUE TITLE
"source activate" causes error in pyenv-which-ext that exits shell

### ISSUE THREAD

**Issue by jimmywan** [yyuu/pyenv#399]
I had a little trouble getting virtualenvwrapper to work properly with conda, but it doesn't sound like that's really a supported configuration by conda, So I tried to use conda envs directly, when this happened.

First setup pyenv version and find activate/conda
```
vagrant@ubuntu-14:~$ PYENV_DEBUG=1
vagrant@ubuntu-14:~$ which activate
/home/vagrant/.pyenv/shims/activate
vagrant@ubuntu-14:~$ less /home/vagrant/.pyenv/shims/activate
vagrant@ubuntu-14:~$ pyenv shell miniconda-3.9.1
vagrant@ubuntu-14:~$ which activate
/home/vagrant/.pyenv/shims/activate
vagrant@ubuntu-14:~$ pyenv which activate
/home/vagrant/.pyenv/versions/miniconda-3.9.1/bin/activate
vagrant@ubuntu-14:~$ which conda
/home/vagrant/.pyenv/shims/conda
vagrant@ubuntu-14:~$ pyenv which conda
/home/vagrant/.pyenv/versions/miniconda-3.9.1/bin/conda
```
So at this point, it looks like I'm correctly using miniconda-3.9.1 and pyenv correctly appears to want to use the activate definition from miniconda-3.9.1.

Next I'll verify that the conda environment is there
```
vagrant@ubuntu-14:~$ conda info --envs
# conda environments:
#
manda                    /home/vagrant/.pyenv/versions/miniconda-3.9.1/envs/manda
root                  *  /home/vagrant/.pyenv/versions/miniconda-3.9.1
```
I've confirmed that conda thinks I have a conda env named "manda"

Now I'll try to activate my environment (http://conda.pydata.org/docs/using/envs.html), but some interactions between conda and the lookup_from_path.bash script causes me to get kicked out of my shell:
```
vagrant@ubuntu-14:~$ source activate manda
++ program=-bash
++ '[' -bash = python ']'
++ export PYENV_ROOT=/home/vagrant/.pyenv
++ PYENV_ROOT=/home/vagrant/.pyenv
++ exec /home/vagrant/.pyenv/libexec/pyenv exec -bash manda
/home/vagrant/.pyenv/plugins/pyenv-which-ext/etc/pyenv.d/which/lookup_from_path.bash: line 38: command: -b: invalid option
command: usage: command [-pVv] command [arg ...]
pyenv: -bash: command not found
Connection to 127.0.0.1 closed.
```

Any ideas?

**Comment by jimmywan** [comment]
This is the relevant part of my ~/.profile
``` bash
if [ -f ~/.bashrc ]; then
   source ~/.bashrc
fi

export PYENV_ROOT="$HOME/.pyenv"
export PATH="$PYENV_ROOT/bin:$PATH"
eval "$(pyenv init -)"
eval "$(pyenv virtualenv-init -)"

export WORKON_HOME=${HOME}/.virtualenvs
export PROJECT_HOME=/cygwin/code
# Disable virtualenvwrapper for now
#pyenv virtualenvwrapper
```

Some relevant environment variables after I set my pyenv version:
```
vagrant@ubuntu-14:~$ env | egrep 'PYENV|PATH'
PYENV_ROOT=/home/vagrant/.pyenv
PYENV_VERSION=miniconda-3.9.1
PYENV_VIRTUALENV_INIT=1
PATH=/home/vagrant/.pyenv/shims:/home/vagrant/.pyenv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
PYENV_SHELL=bash
```

**Comment by jimmywan** [comment]
Should I move this issue from pyenv to pyenv-virtualenv?

**Comment by yyuu** [comment]
@jimmywan Looks like configuration problem with pyenv-virtualenv + anaconda/miniconda. Please move issue to pyenv-virtualenv.

**Comment by jimmywan** [comment]
Moved to pyenv-virtualenv:
https://github.com/yyuu/pyenv-virtualenv/issues/88

**Comment by n8henrie** [comment]
If anyone ends up here after Googling (like I did), I just ran into the same error, but not related to `pyenv-virtualenv` in any way to my knowledge:

```
/home/raspi/.pyenv/libexec/pyenv-which: line 43: command: -b: invalid option
command: usage: command [-pVv] command [arg ...]
pyenv: -bash: command not found
Connection to raspi.local closed.
```

Every time I sourced my `.bashrc` (with my pyenv init stuff), I'd get kicked off the SSH connection, as seems to be the case with the OP here. The really odd thing was I hadn't changed my `.bashrc` setup.

I'm not exactly sure what caused the error, but I figured I'd try a `pyenv rehash`, and I got *another* error:

```
$ pyenv rehash
pyenv: cannot rehash: /home/raspi/.pyenv/shims/.pyenv-shim exists
```

After inspecting and backing up that file, I tried deleting it: `rm /home/raspi/.pyenv/shims/.pyenv-shim` and afterwards both errors were fixed. Not sure why.
